AK/String.cpp

```diff
@@ -196,4 +196,58 @@ String String::repeated(char ch, int count)
     return *impl;
 }
 
+bool String::matches(const String& mask, CaseSensitivity case_sensitivity) const
+{
+    if (case_sensitivity == CaseSensitivity::CaseInsensitive) {
+        String this_lower = this->to_lowercase();
+        String mask_lower = mask.to_lowercase();
+        return this_lower.match_helper(mask_lower);
```

Comment:
In the glorious future we could have a version of this that doesn't create a bunch of new strings just for the comparison.

Reply:
Yes, agreed. match_helper used to operate on the lowercase version of the mask/string inline, I was just too lazy to do so now :)